### PR TITLE
[BEAM-105] Only remove window from active window set if it is still active

### PR DESCRIPTION
**Pull request by mshields822**
Will start testing now.

### CODE REVIEW

**Comment by mshields822** [comment]
I'd strongly encourage pulling this for 1.5.
R=@davorbonaci

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/util/ReduceFnRunnerTest.java [hunk 1]
```diff
@@ -687,6 +687,44 @@ public void testMergeBeforeFinalizing() throws Exception {
...
+    assertThat(output.get(0),
+        isSingleWindowedValue(containsInAnyOrder(1, 10),
+            1, // timestamp
+            1, // window start
+            20)); // window end
+    assertThat(
+        output.get(0).getPane(),
+        equalTo(PaneInfo.createPane(true, true, Timing.ON_TIME, 0, 0)));
+  }
+
```
Comment (davorbonaci) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/util/ReduceFnRunnerTest.java]: remove extra newline

**Comment by davorbonaci** [comment]
LGTM, but I'd like a second pair of eyes.

@kennknowles or @bjchambers, could you please take a quick peek?

File: sdk/src/test/java/com/google/cloud/dataflow/sdk/util/ReduceFnRunnerTest.java (hunk 1 above)
Comment (mshields822) [inline sdk/src/test/java/com/google/cloud/dataflow/sdk/util/ReduceFnRunnerTest.java]: Done.

File: sdk/src/main/java/com/google/cloud/dataflow/sdk/util/MergingActiveWindowSet.java [hunk 2]
```diff
@@ -195,6 +191,7 @@ public void addActive(W window) {
 
   @Override
   public void remove(W window) {
+    Preconditions.checkState(isActive(window), "Window %s is not active", window);
```
Comment (kennknowles) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/util/MergingActiveWindowSet.java]: Seems reasonable, but could we just make removal idempotent?

Comment (mshields822) [inline sdk/src/main/java/com/google/cloud/dataflow/sdk/util/MergingActiveWindowSet.java]: There are other calls to remove and I want to assert they are only removing active windows.

**Comment by kennknowles** [comment]
Noting that the #36 meant to be auto-closed was from a different repo, so go ahead and re-open this or open a new one.

**Pull request closed without merging**